buku

```diff
@@ -1143,9 +1357,20 @@ class BukuDb:
             LOGERR('Records: %d, processed: %d !!!', recs, processed['value'])
 
         cond.release()
-        self.conn.commit()
+        if delay_delete:
+            self.conn.commit()
+        else:
+            self.commit_delete()
         return True
 
+    def commit_delete(self, apply: bool = True):
+        """Commit delayed delete commands."""
+        if apply and self._to_delete is not None:
+            for id in sorted(set(self._to_delete), reverse=True):
```

Comment:
Going right-to-left is necessary to prevent invalidating indices before they're processed